reference-implementation/lib/readable-stream.js

```diff
@@ -397,9 +417,44 @@ function ErrorReadableStream(stream, e) {
   stream._storedError = e;
   stream._state = 'errored';
 
-  if (IsReadableStreamLocked(stream) === true) {
-    return ReleaseReadableStreamReader(stream._reader);
+  const reader = stream._reader;
+
+  if (reader === undefined) {
+    return undefined;
   }
+
+  for (const { _reject } of reader._readRequests) {
+    _reject(e);
+  }
+  reader._readRequests = [];
+
+  ReleaseReadableStreamReader(reader);
```

Comment:
This call is placed here so that this would be parallel with RBS. In RBS I'm calling this right after flushing read requests because whether we release the reader or not depends on emptiness of read requests queue.